### ISSUE TITLE
Expose a getter for the global window

### ISSUE THREAD

**Issue by fitzgen** [rustwasm/wasm-bindgen#659]
I think we could do something like

```rust
#[wasm_bindgen]
extern {
    #[wasm_bindgen(getter)]
    pub fn window() -> Window;
}
```

inside `crates/web-sys/src/lib.rs` and consider that the entry point to all globally exposed properties and methods.

**Comment by alexcrichton** [comment]
Currently the way to do this would look like:

```rust
#[wasm_bindgen]
extern {
    static window: Window;
}

// use `window`
```

but importing via a `static` is a little sketchy and I think I'd prefer to get something like a bare `getter` you've mentioned @fitzgen to work, that makes more sense to me.

**Comment by derekdreery** [comment]
Out of curiosity, why is using a `static` sketchy?

**Comment by alexcrichton** [comment]
Ah just a personal gut feeling! Using a `static` creates a different type (a [`JsStatic`](https://github.com/rustwasm/wasm-bindgen/blob/a4bf239effe681c2dd3883d7f22880e3bf709bae/src/lib.rs#L496-L555)) which has a different API to learn and can be non-obvious. Using a function seems much better to me as it's more isolated and fits well with the rest of the API.

**Comment by eminence** [comment]
Another benefit of using a function is that it's easier to search for in the docs

**Comment by fitzgen** [comment]
We talked about this in the WG meeting today. We decided that when an interface has the `[Global]` attribute, then we should generate free functions for each of its operations and members without the `&self` parameter.

This didn't come up at the meeting, but since there are multiple interfaces with `[Global]` that represent workers, main window, worklets, etc... we *probably* want these to be static methods for disambiguation and context.
